app/src/androidTest/java/com/github/sdp_begreen/begreen/viewModels/MeetingFragmentViewModelTest.kt

```diff
@@ -0,0 +1,351 @@
+import com.github.sdp_begreen.begreen.firebase.meetingServices.MeetingService
+import com.github.sdp_begreen.begreen.models.CustomLatLng
+import com.github.sdp_begreen.begreen.models.User
+import com.github.sdp_begreen.begreen.models.Meeting
+import com.github.sdp_begreen.begreen.rules.CoroutineTestRule
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+import org.hamcrest.Matchers.equalTo
+import org.hamcrest.Matchers.`is`
+import org.hamcrest.Matchers.nullValue
+import org.junit.Before
+import org.junit.BeforeClass
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.koin.dsl.module
+import org.koin.test.KoinTestRule
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(AndroidJUnit4::class)
+@MediumTest
+class MeetingFragmentViewModelTest {
+
+    companion object {
+        private val meetingService: MeetingService = mock(MeetingService::class.java)
+        private val participantService: MeetingParticipantService =
+            mock(MeetingParticipantService::class.java)
+
+        private val initiallyConnectedUser = User("123456", 10, "User 1")
+        private val currentUser: MutableStateFlow<User?> = MutableStateFlow(initiallyConnectedUser)
+
+        private val meetings = listOf(
+            Meeting(
+                "m1",
+                "John",
+                "Downtown Cleanup",
+                "Let's meet up and clean up the downtown area",
+                1718445600000,
+                1718467200000,
+                CustomLatLng(37.7749, -122.4194),
+                CustomLatLng(37.7749, -122.4194)
+            ),
+            Meeting(
+                "m2",
+                "Sarah",
+                "Park Cleanup",
+                "Let's meet up and clean up the local park",
+                1718186400000,
+                1718208000000,
+                CustomLatLng(37.7739, -122.4312),
+                CustomLatLng(37.7739, -122.4312)
+            ),
+            Meeting(
+                "m3",
+                "Tom",
+                "Beach Cleanup",
+                "Let's meet up and clean up the local beach",
+                1710237600000,
+                1710252000000,
+                CustomLatLng(46.517355, 6.628854),
+                CustomLatLng(46.517355, 6.628854)
+            ),
+        )
+
+        private val meetingsFlow = MutableStateFlow(meetings)
+
+        // supposed to represent a Channel where both add participant and
+        // remove participant can be send through
+        val addRemoveParticipantChannel = Channel<String>(1)
+
+        @BeforeClass
+        @JvmStatic
+        fun setUpAuth() {
+            runTest {
+
+                // Initial setup of getAllMeetings
```

Comment:
It seems to me that some of the code below is duplicated from `MeetingsFragmentTest`

Reply:
You are right, there is some duplicated code indeed. I started by thinking of a way to not duplicate it, but then I thought that as this specific setup will only be used in these two test classes, it would not make a lot a sens to extract this code in an `initTestUtil` class for example. I am not very happy with this duplication either, but I don't really know what would be the best solution to get ride of it. Would you have any suggestion that I could apply in this scenario ?

Reply:
If this ducplication only appears twice, perhaps it is not necessary to extract it into a utility function or implement another complex solution, such as having a shared parent class for `MeetingFragments`. Unless there is additional shared code that needs to be extracted, it might be best to leave it as it is.

Reply:
Ok, thank you very much for your answer. So I will leave it that way for now, but keep it in mind if later I need the same kind of setup, so that I can extract the common code.